docs/tutorials/using_apis/connectivity/tls_tutorial.md

```diff
@@ -0,0 +1,173 @@
+
+const char SSL_CA_PEM[] = /* your certificate, see above */ "";
+ 
+int main (void) {
+
+// Get a network interface
+NetworkInterface *network = NetworkInterface::get_default_instance();
+if (network->connect() != 0) {
+    printf("Could not connect to the network...\n");
+    return 1;
+}
+ 
+nsapi_error_t r;
+ 
+// setting up TLS socket
+TLSSocket* socket = new TLSSocket();
+if ((r = socket->open(network)) != NSAPI_ERROR_OK) {
+    printf("TLS socket open failed (%d)\n", r);
+    return 1;
+}
+if ((r = socket->set_root_ca_cert(SSL_CA_PEM)) != NSAPI_ERROR_OK) {
+    printf("TLS socket set_root_ca_cert failed (%d)\n", r);
+    return 1;
+}
+if ((r = socket->connect("os.mbed.com", 443)) != NSAPI_ERROR_OK) {
+    printf("TLS socket connect failed (%d)\n", r);
+    return 1;
+}
+
+}
+```
+
+This now makes the request and returns the content of the file.
+
+```
+Setting up TLS socket...
+[INFO][TLSW]: mbedtls_ssl_conf_ca_chain()
+[INFO][TLSW]: mbedtls_ssl_config_defaults()
+[INFO][TLSW]: mbedtls_ssl_conf_authmode()
+[INFO][TLSW]: mbedtls_ssl_conf_rng()
+[INFO][TLSW]: mbedtls_ssl_setup()
+[INFO][TLSW]: Starting TLS handshake with os.mbed.com
+[INFO][TLSW]: TLS connection to os.mbed.com established
+ 
+[DBG ][TLSW]: Server certificate:
+    cert. version     : 3
+    serial number     : 03:56:D4:79:41:63:31:CA:E0:56:06:61
+    … snip ...
+ 
+[INFO][TLSW]: Certificate verification passed
+```
+
+You can also look at the [complete example](https://github.com/janjongboom/mbed-simulator/blob/6e9f45c61fb10e1983668e777d68e939247b377d/demos/tlssocket/main.cpp) for more details.
+
+<span class="notes">**Note:** If you come across a message saying the signature's algorithm is missing: <br>
+```
+[ERR ][TLSW]: mbedtls_x509_crt_parse() failed: -0x262e (-9774): X509 - Signature algorithm (oid) is unsupported : OID - OID is not found
+```
+then you may need to add the following line in `target_overrides` section of your `mbed_app.json` file:<br>
+```json
+    "target_overrides": {
+        "*": {
+            "target.macros_add" : ["MBEDTLS_SHA1_C"]
+        }
+    }
+```
+</span>
+
+### Providing client's certificate and key
+
+The CA certification only allows you to verify the server's authenticity.
+
+The TLS handshake can also perform the second stage of verification where the client's certificate is verified to confirm its authenticity.
+
+To achieve this, you must call the TLS socket's `set_client_cert_key()` function, which stores your key and certificate pair within the TLS socket. This may be required in cases where the server needs to verify the client's identity for secure communication, for example when your IoT device wants to send (and not just receive) some data to the server.
+
+The mechanism of providing the client's key and certificate to TLS socket is the same as in case of CA certificate. You can find examples of how and when to perform this operation in the [cloud demo applications](https://github.com/coisme/Mbed-to-AWS-IoT).
```

Comment:
We have a policy that we don't include code from individual user accounts in our public documentation. Do we have example in ARMmbed?

Reply:
All our cloud demo application live in the domain of @coisme 's account. I do not know if there is any plan to move them to ARMmbed. As far as I understand, despite not being added to ARMmbed, this is an officially supported ARM demo for cloud connectivity. It is also a very nice and useful example to show case the function...
Is it possible to make an exception to the policy and keep the link to the cloud demo project?

Reply:
I have no plan to move them to ARMmbed at the moment. I have no problem with moving them to under ARMmbed, but I cannot engage in maintenance.

Reply:
Because this example is referenced from our official documentation, it needs to live in ARMmbed. Please move it there.

Reply:
@coisme , are you able to move the examples, as requested?

Reply:
Following @SeppoTakalo 's suggestion I dropped the example mention from the docs.